File: mTemp-API/mTemp-API/Domain/Models/TemperatureMeasurement.cs
namespace mTemp_API.Domain.Models
{

	public class TemperatureMeasurement
	{
		public int Id { get; set; } // GUID - Unique identifier.
		public decimal MeasuredTemperature { get; set; } //  The numerical value of the temperature reading.
		public string MeasuredMethod { get; set; } = string.Empty; // The method used for taking the temperature.
		public  DateTime Timestamp { get; set; } // When the measurement was recorded.
		public int? PatientId { get; set; } // Foreign key to the Patient table.

	}
}

File: mtemp-webapp/src/models/TemperatureMeasurement.ts
export class TemperatureMeasurement {
    id: number | undefined;
    measuredTemperature: number;
    measuredMethod: string;
    timestamp: number | undefined;
    patientId: number | undefined;


  constructor(id: number | undefined = undefined, measuredTemperature: number, measuredMethod: string, timestamp: number | undefined = undefined, patientId: number | undefined = undefined) {
    this.id = id;
    this.measuredTemperature = measuredTemperature;
    this.measuredMethod = measuredMethod;
    this.timestamp = timestamp;
    this.patientId = patientId;
  }

  static fromJSON(json: any): TemperatureMeasurement {
    return new TemperatureMeasurement(json.id, json.measuredTemperature, json.measuredMethod, json.timestamp, json.patientId);
  }


}